File: Backend/app/models/user_model.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, Boolean
from sqlalchemy.orm import relationship
from datetime import datetime, timezone
from app.db.base import Base

class User(Base):
    __tablename__ = "users"

    id = Column(Integer, primary_key=True)
    role_id = Column(Integer, ForeignKey("roles.id"), nullable=False)

    name = Column(String, nullable=False)
    last_name = Column(String, nullable=False)
    email = Column(String, unique=True, nullable=False)
    password = Column(String, nullable=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = Column(DateTime, onupdate=lambda: datetime.now(timezone.utc))
    deleted_at = Column(DateTime, nullable=True)

    is_active = Column(Boolean, default=True)
    gps_status = Column(String, default="none") # "none", "pending", "approved"
    gps_imei = Column(String, nullable=True)

    role = relationship("Role", back_populates="users")
    pets = relationship("Pet", back_populates="publisher")

File: FrontendV2/src/app/shared/models/user.model.ts
export interface Role {
  id: number;
  name: string;
}

export interface User {
  // Campos obligatorios
  id: number;
  role_id: number;
  name: string;
  last_name: string;
  email: string;
  created_at: string;

  // Campos opcionales
  updated_at?: string | null;
  deleted_at?: string | null;
  is_active?: boolean;

  // Relación
  role?: Role;
}

export interface AuthResponse {
  access_token: string;
  user: User;
}
